spec/xml_security_spec.rb

```diff
@@ -41,10 +41,22 @@ module SamlIdp
           )
         end
 
+        it 'raises require SHA256 error' do
+          response = Base64.decode64(response_document)
+          response.sub!('<ds:DigestValue>pJQ7MS/ek4KRRWGmv/H43ReHYMs=</ds:DigestValue>',
+                        '<ds:DigestValue>b9xsAXLsynugg3Wc1CI3kpWku+0=</ds:DigestValue>')
+          document = XMLSecurity::SignedDocument.new(response)
+          base64cert = document.elements['//ds:X509Certificate'].text
+          expect { document.validate_doc(base64cert, false) }.to(
+            raise_error(SamlIdp::XMLSecurity::SignedDocument::ValidationError, 'All signatures must use RSA SHA-256')
+          )
+        end
```

Comment:
i really want to fix these tests (and `spec/support/security_helpers.rb` but this scope has already gotten expanded so left a TODO in `security_helpers.rb`